src/utils/EDA.py

```diff
@@ -0,0 +1,177 @@
+# Libraries 
+import pandas as pd
+import numpy as np 
+
+# Plotting
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+
+def plt_missing_data(data, title):
+    '''
+    Plots a bar chart showing the percentage of missing values for each column.
+    
+    Args:
+        data (pandas.DataFrame): The DataFrame to plot.
+        str (str): A string to include in the chart title indicating the intended cleaning action.
+
+    Returns:
+        None. Displays a bar chart using matplotlib.
+    '''
+    # Calculate the percentage of missing values for each column
+    missing_percentages = data.isnull().mean() * 100
+
+    # Filter the columns that have missing values
+    missing_columns = missing_percentages[missing_percentages > 0]
+
+    # Sort the columns by percentage of missing values
+    missing_columns = missing_columns.sort_values(ascending=False)
+
+    # Plot the percentage of missing values for each column
+    plt.bar(missing_columns.index, missing_columns.values)
+    plt.title(title)
+    plt.xlabel('Column Name')
+    plt.ylabel('Percentage of Missing Values')
+    plt.xticks(rotation=90)
+    plt.show()
+
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+def create_histogram(data, bin=30, orientation='vertical', sort=True, xlabel='', ylabel='', title=''):
+    """
+    Creates a histogram of the given data.
+
+    Args:
+        data (list or numpy array): The data to create a histogram.
+        bin (int): The num of bins
+        orientation (str, optional): The orientation of the histogram, either 'vertical' (default) or 'horizontal'.
+        sort (bool, optional): Whether to sort the values in data before creating the histogram. Defaults to True.
+        xlabel (str, optional): The label for the x-axis.
+        ylabel (str, optional): The label for the y-axis.
+        title (str, optional): The title of the histogram.
+
+    Returns:
+        None
+    """
+    if sort:
+        data = np.sort(data)
+
+    plt.hist(data, bins=bin, orientation=orientation)
+    plt.xlabel(xlabel)
+    plt.ylabel(ylabel)
+    plt.title(title)
+    plt.show()
+
+
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+
+def create_stacked_histogram(df, column1, column2, bins=10, sort=True, title="", xlabel="", ylabel="Frequency", orientation="vertical"):
```

Comment:
Like the previous, could you add the type hint for returned value and the arguments except for arguments that have a default value?
Also, the title and xlabel inputs are not optional arguments...